sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1994,13 +1994,17 @@ case class MakeTimestamp(
       day: Int,
       hour: Int,
       min: Int,
-      secAndNanos: Decimal,
+      secAndMicros: Decimal,
       zoneId: ZoneId): Any = {
     try {
-      val secFloor = secAndNanos.floor
-      val nanosPerSec = Decimal(NANOS_PER_SECOND, 10, 0)
-      val nanos = ((secAndNanos - secFloor) * nanosPerSec).toInt
-      val seconds = secFloor.toInt
+      assert(secAndMicros.scale == 6,
```

Comment:
I doubt that this is the right place to ask, but I tried to Google this but didn't find anything. I was just wondering what does `assert(secAndMicros.scale == 6` do exactly? I am assuming that it changes the Decimal's scale to 6, but what does that exactly help with? And is the added tail just a bunch of zeros? For instance does 3.14 become 3.140000?
Thanks!

Reply:
> I am assuming that it changes the Decimal's scale to 6

This assert doesn't change the scale, it just reads it.

> And is the added tail just a bunch of zeros? For instance does 3.14 become 3.140000?

It shifts the decimal point. For example, if you have 3.14 with precision 6 and scale 3 that means 003.140. If you set
- scale to 0, it becomes 3140
- scale to 4 -> 0.314

Reply:
Awesome, thank you so much!